Sources/Operators/CombineLatestMany.swift

```diff
@@ -18,17 +18,9 @@ public extension Publisher {
     ///
     /// - returns: A type-erased publisher with value events from `self` and each of the inner publishers `combineLatest`’d
     /// together in an array.
-    func combineLatest<Others: Collection>(with others: Others)
-        -> AnyPublisher<[Output], Failure>
-        where Others.Element: Publisher, Others.Element.Output == Output, Others.Element.Failure == Failure {
-        let seed = map { [$0] }.eraseToAnyPublisher()
-
-        return others.reduce(seed) { combined, next in
-            combined
-                .combineLatest(next)
-                .map { $0 + [$1] }
-                .eraseToAnyPublisher()
-        }
+    func combineLatest<Others: Collection>(with others: Others) -> AnyPublisher<[Output], Failure>
+        where Others.Element == Self, Others.Element.Output == Output, Others.Element.Failure == Failure {
+            ([self] + others).combineLatest()
```

Comment:
@freak4pc @jasdev How important are this func and the next? This PR changes the typing slightly to be more restrictive (self and the combined-with publishers must share a type, instead of simply both being Publishers that share Output & Failure types), but apparently there isn't dedicated test coverage for them, and double-so for not having test coverage of different types of publishers being combined :)
The code-paths that would get here from the old impl would already enforce this same-type requirement (b/c "self" here is in the Collection there), so this seems fine to me.

Reply:
We'd want to keep the existing surface area (meaning existing funds and return types), as otherwise it would be breaking

Reply:
Fair enough. Figured out how to preserve the previous signatures.